src/highcharts-ng.js

```diff
@@ -50,26 +51,74 @@ if (typeof module !== 'undefined' && typeof exports !== 'undefined' && module.ex
       }, 0, false);
     };
 
+        this.removeUnlinkedObjects = function(mergedConfig) {
+            /*
+                Removes unlinked objects, items that have been removed in the config,
+                but not yet removed from the HighChart object
+            */
+
+            //First check to see if there are any yAxis that need to be removed
+            //If a series is linked to the yAxis, it will be removed by HighCharts
+
+            if (mergedConfig.yAxis && Array.isArray(mergedConfig.yAxis)) {
```

Comment:
Looks good. I wonder if we could factor out the common logic for yAxis and Series. Then we could easily add xAxis to the same logic. It would probably be sufficient to check the id is not one of ['navigator-y-axis', 'navigator-x-axis', 'highcharts-navigator-series']